frontend/src/features/codes/OwnersList.tsx

```diff
@@ -0,0 +1,113 @@
+import {IconCrossCircle, IconSearch, TextInput} from "hds-react";
+import {useCallback, useRef, useState} from "react";
+import {useGetOwnersQuery} from "../../app/services";
+import {ModifyOwnerModal, QueryStateHandler} from "../../common/components";
+import {IFilterOwnersQuery, IOwner} from "../../common/schemas";
+
+const OwnerResultList: React.FC<{params: IFilterOwnersQuery}> = ({params}) => {
+    const MIN_LENGTH = 2; // Minimum length before applying filter
+    const MAX_ROWS = 12; // Maximum rows to show
+    if (params.name?.length && params.name?.length < MIN_LENGTH) {
+        delete params.name;
+    }
+    if (params.identifier?.length && params.identifier?.length < MIN_LENGTH) {
+        delete params.identifier;
+    }
+    // get the data
+    const {data, error, isLoading} = useGetOwnersQuery({...params, limit: MAX_ROWS});
+
+    // state for the modal
+    const [isModifyOwnerModalOpen, setIsModifyOwnerModalOpen] = useState(false);
+    const [owner, setOwner] = useState<IOwner | undefined>(undefined);
+
+    // action for the row click
+    const editFn = (ownerEdited: IOwner) => {
+        setIsModifyOwnerModalOpen(true);
+        setOwner(ownerEdited);
+    };
+    return (
+        <QueryStateHandler
+            data={data}
+            error={error}
+            isLoading={isLoading}
+        >
+            <div className="list-headers">
+                <div>Nimi</div>
+                <div>Henkilö- tai Y-tunnus</div>
+            </div>
+            <ul className="results-list">
+                {data?.contents.map((owner: IOwner) => (
+                    <div
+                        key={owner.identifier}
+                        className="results-list__item"
+                        onClick={(e) => {
+                            e.preventDefault();
+                            editFn(owner);
+                        }}
+                    >
+                        <span>{owner.name}</span>
+                        <span>{owner.identifier}</span>
+                    </div>
+                ))}
+            </ul>
+            <span className="list-footer">
+                Näytetään {data?.page.size}/{data?.page.total_items} hakutulosta
+            </span>
+            <ModifyOwnerModal
+                owner={owner as IOwner}
+                isVisible={isModifyOwnerModalOpen}
+                setIsVisible={setIsModifyOwnerModalOpen}
+            />
+        </QueryStateHandler>
+    );
+};
+
+export default function OwnersList() {
+    // search strings
+    const [filterParams, setFilterParams] = useState<IFilterOwnersQuery>({name: "", identifier: ""});
+
+    // focus the field when clicking its cross circle icon
```

Comment:
Ooh, this is a nice one 😄 
You should consider adding this to `FilterTextInputField` also!

Reply:
To add this feature in FilterTextInputField, it should be changed to a controlled component, that would have an impact to the other components using it (e.g. queries that use min length). So I think this change is a bigger one and should be done in another PR.